isaaclab_arena/tasks/assembly_task.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) 2025, The Isaac Lab Arena Project Developers (https://github.com/isaac-sim/IsaacLab-Arena/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+from dataclasses import MISSING, dataclass
+
+import isaaclab.envs.mdp as mdp_isaac_lab
+from isaaclab.envs.common import ViewerCfg
+from isaaclab.envs.mimic_env_cfg import MimicEnvCfg
+from isaaclab.managers import EventTermCfg, SceneEntityCfg, TerminationTermCfg
+from isaaclab.utils import configclass
+
+import isaaclab_arena_environments.mdp as mdp
+from isaaclab_arena.assets.asset import Asset
+from isaaclab_arena.metrics.metric_base import MetricBase
+from isaaclab_arena.metrics.object_moved import ObjectMovedRateMetric
+from isaaclab_arena.metrics.success_rate import SuccessRateMetric
+from isaaclab_arena.tasks.events import randomize_poses_and_align_auxiliary_assets
+from isaaclab_arena.tasks.task_base import TaskBase
+from isaaclab_arena.tasks.terminations import objects_in_proximity
+from isaaclab_arena.utils.cameras import get_viewer_cfg_look_at_object
+
+
+@dataclass
+class TerminationConfig:
+    max_x_separation: float = 0.020
+    max_y_separation: float = 0.020
+    max_z_separation: float = 0.020
+
+
+class AssemblyTask(TaskBase):
+    """
+    Assembly task where an object needs to be assembled with a base object, like peg insert, gear mesh, etc.
+    """
+
+    def __init__(
+        self,
+        fixed_asset: Asset,
+        held_asset: Asset,
+        auxiliary_asset_list: list[Asset],
+        background_scene: Asset,
+        episode_length_s: float | None = None,
+        termination_cfg: TerminationConfig = TerminationConfig(
+            max_x_separation=0.020, max_y_separation=0.020, max_z_separation=0.020
+        ),
```

Comment:
Suggestion to just use a `tuple` here.

Introduction of another class `TerminationConfig`, is kind of confusing because we have several classes called similar things in this file: `TerminationTermCfg`, `TerminationsCfg`